scene/gui/text_edit.cpp

```diff
@@ -623,7 +623,7 @@ void TextEdit::_notification(int p_what) {
 						}
 					}
 
-					if (get_caret_column(caret) > 0) {
+					if (get_caret_column(caret) > 0 && get_caret_column(caret) <= text[get_caret_line(caret)].length()) {
```

Comment:
If @Paulb23 doesn't like that approach, maybe we can make this an `ERR_FAIL_COND` instead, so it still fixes the crash but points that there's something fishy?